scripts/strip.py

```diff
@@ -0,0 +1,278 @@
+    ]
+
+
+@dataclass
+class ElfShdr(ElfStruct):
+    """Abstraction for structs Elf32_Shdr and Elf64_Shdr"""
+    sh_name: int
+    sh_type: int
+    sh_flags: int
+    sh_addr: int
+    sh_offset: int
+    sh_size: int
+    sh_link: int
+    sh_info: int
+    sh_addralign: int
+    sh_entsize: int
+
+
+@dataclass
+class Elf32Shdr(ElfShdr):
+    """struct Elf32_Shdr"""
+    FORMAT = [
+        ("I", "sh_name"),
+        ("I", "sh_type"),
+        ("I", "sh_flags"),
+        ("I", "sh_addr"),
+        ("I", "sh_offset"),
+        ("I", "sh_size"),
+        ("I", "sh_link"),
+        ("I", "sh_info"),
+        ("I", "sh_addralign"),
+        ("I", "sh_entsize")
+    ]
+
+
+@dataclass
+class ElfRelx(ElfStruct):
+    """Abstraction for structs Elf32_Rel, Elf64_Rel, Elf32_Rela, Elf64_Rela"""
+    r_offset: int
+    r_info: int
+    r_addend: int = 0
+
+@dataclass
+class Elf32Rel(ElfRelx):
+    """struct Elf32_Rel"""
+    FORMAT = [
+        ("I", "r_offset"),
+        ("I", "r_info")
+    ]
+
+
+class ElfFixedSizeTable:
+    offset: int
+    size: int
+    entrySize: int
+    header: Type[ElfStruct]
+    parser: "ElfParser"
+
+    def __init__(self, p: "ElfParser"):
+        self.parser = p
+
+    def __iter__(self):
+        for off in range(self.offset, self.offset + self.size, self.entrySize):
+            assert self.entrySize == self.header.get_size()
+            yield self.parser.read_struct(self.header, off), off
+
+
+class ElfSectionTable(ElfFixedSizeTable):
+    header: Type[ElfShdr]
+
+    def __init__(self, e: ElfEhdr, p: "ElfParser"):
+        super().__init__(p)
+        self.header = {EiClass.ELFCLASS32: Elf32Shdr}[p.ident.get_class()]
+        self.offset = e.e_shoff
+        self.size = e.e_shnum * e.e_shentsize
+        self.entrySize = e.e_shentsize
+
+
+class ElfRelocationTable(ElfFixedSizeTable):
+    header: Type[ElfRelx]
+
+    def __init__(self, s: ElfShdr, p: "ElfParser"):
+        super().__init__(p)
+        self.header = {(EiClass.ELFCLASS32, ShType.SHT_REL): Elf32Rel}[p.ident.get_class(), ShType(s.sh_type)]
+        self.offset = s.sh_offset
+        self.size = s.sh_size
+        self.entrySize = s.sh_entsize
+
+
+class ElfParser:
+    data: BytesIO
+    ident: ElfEident
+    header: ElfEhdr
+
+    def __init__(self, b: BytesIO):
+        self.data = b
+        self.ident = ElfEident.parse(self.data.read(16))
+        header_type = {EiClass.ELFCLASS32: Elf32Ehdr}[self.ident.get_class()]
+        header = self.read_struct(header_type, 16)
+        assert isinstance(header, ElfEhdr)
+        self.header = header
+        self.read_struct(header_type, 16)
+
+    def read_struct(self, st: Type[ElfStruct], offset: int = 0):
+        self.data.seek(offset)
+        data = self.data.read(st.get_size())
+        return st.parse(data, self.ident.get_endianness())
+
+    def write_struct(self, st: ElfStruct, offset: int = None):
+        self.data.seek(offset)
+        self.data.write(st.serialize(self.ident.get_endianness()))
```

Comment:
Offset default value should be same for both functions.  None in my example was intended as "use current offset without seek".  

Maybe remove default value?